### PR TITLE
pkp/pkp-lib#8670 Delete all related data and files when submission fi…

### PR DESCRIPTION
**Pull request by NateWr**
…le deleted

### CODE REVIEW

File: classes/submissionFile/DAO.php [hunk 1]
```diff
@@ -306,33 +307,25 @@ public function getByBestId(
...
-    /**
-     * Remove a specific file assignment from a review round.
-     */
-    public function deleteReviewRoundAssignment(int $submissionFileId): void
-    {
-        // Remove currently assigned review files.
-        DB::table('review_round_files')->where([
-            'submission_file_id' => (int) $submissionFileId
-        ])->delete();
+        DB::table('...')->updateOrInsert(
```
Comment (Vitaliy-1) [inline classes/submissionFile/DAO.php:316]: `review_round_files` table?

File: classes/submissionFile/Repository.php [hunk 2]
```diff
@@ -422,7 +424,105 @@ public function copy(SubmissionFile $submissionFile, int $toFileStage, ?int $rev
...
+        $noteDao = DAORegistry::getDAO('NoteDAO'); /** @var NoteDAO $noteDao */
+        $noteDao->deleteByAssoc(Application::ASSOC_TYPE_SUBMISSION_FILE, $submissionFile->getId());
+
+        // Update tasks
+        $notificationMgr = new NotificationManager();
+        switch ($submissionFile->getData('fileStage')) {
+            case SubmissionFile::SUBMISSION_FILE_REVIEW_REVISION:
+                $authorUserIds = [];
+                $stageAssignmentDao = DAORegistry::getDAO('StageAssignmentDAO'); /** @var StageAssignmentDAO $stageAssignmentDao */
+                $submitterAssignments = $stageAssignmentDao->getBySubmissionAndRoleIds($submissionFile->getData('submissionId'), [ROLE_ID_AUTHOR]);
```
Comment (Vitaliy-1) [inline classes/submissionFile/Repository.php:450]: Role::ROLE_ID_AUTHOR

File: classes/submissionFile/Repository.php [hunk 3]
```diff
@@ -422,7 +424,105 @@ public function copy(SubmissionFile $submissionFile, int $toFileStage, ?int $rev
...
+            Application::get()->getRequest(),
+            $submissionFile,
+            SubmissionFileEventLogEntry::SUBMISSION_LOG_FILE_DELETE,
+            'submission.event.fileDeleted',
+            [
+                'fileStage' => $submissionFile->getData('fileStage'),
+                'sourceSubmissionFileId' => $submissionFile->getData('sourceSubmissionFileId'),
+                'submissionFileId' => $submissionFile->getId(),
+                'submissionId' => $submissionFile->getData('submissionId'),
+                'username' => $user ? $user->getUsername() : null,
```
Comment (Vitaliy-1) [inline classes/submissionFile/Repository.php:521]: Just a reminder that we can start using null safe operator from 3.4 as php < 8.0 isn't supported

File: classes/submissionFile/DAO.php (hunk 1 above)
Comment (NateWr) [reply classes/submissionFile/DAO.php:316]: :scream: yikes!

**Pull request merged**